src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/ProxyHeaderModule.cs

```diff
@@ -55,12 +63,12 @@ public void Init(HttpApplication context)
             context.BeginRequest += (s, e) =>
             {
                 var request = ((HttpApplication)s).Context.Request;
-                UseOptions(values, request.Headers, request.ServerVariables);
+                UseOptions(values, request.Headers, request.ServerVariables, request);
             };
         }
     }
 
-    public void UseHeaders(NameValueCollection requestHeaders, NameValueCollection serverVariables)
+    public void UseHeaders(NameValueCollection requestHeaders, NameValueCollection serverVariables, HttpRequest? request = null)
```

Comment:
this class is tested with the public methods (not with Init as HttpApplication cannot be mocked)

Reply:
so tests will probably fail :)

Reply:
So I wasn't planning on adding unit tests for this, once we have playwright setup I could verify this behavior via the contact page showing the correct request.url for the coreapp and the app.

Reply:
The tests use null request which just no-op the reflection

Reply:
I was referring to the existing tests. Do they still pass? This method was designed to be fully testable (hence, passing the headers/server variables rather than the request).

Reply:
does it need to be called there then? especially since both pathways need it, maybe call it after returning and still in the BetginRequest callback,

ie

```csharp
            context.BeginRequest += (s, e) =>
            {
                var request = ((HttpApplication)s).Context.Request;
                UseOptions(values, request.Headers, request.ServerVariables);
                reflectionHelper.SetHostName(request.ServerVariables[Host]);
            };
```

Reply:
Yeah the existing methods pass fine